dataset/st_dataset.py

```diff
@@ -151,4 +151,21 @@ def concatenate_dataset_dicts(dataset_dicts: List[DatasetDict]) -> DatasetDict:
 # Param: DatasetDict
 # rtype: DatasetInfo 
 def get_dataset_info(data_dict: DatasetDict) -> DatasetInfo:
-    return data_dict["train"].info
\ No newline at end of file
+    return data_dict["train"].info
+
+
+
+def load_dataset_safe(*args, **kwargs) -> Dataset:
+
+    tries = 3
+    for i in range(tries):
+        try:
+            ds = load_dataset(*args, *kwargs)
+        except :
+            if i < tries - 1: # i is zero indexed
+                continue
+            else:
+                sys.exit("Wait for a mninute and attempt downloading the dataset again. The server hosting the dataset occassionally times out.")
```

Comment:
You could do `raise RuntimeError("message")` and `return None` instead of simply exiting, which is a bit too drastic imo.